keycloak-user-api/src/main/java/io/enmasse/user/keycloak/KeycloakUserApi.java

```diff
@@ -584,9 +584,9 @@ static User buildUser(UserRepresentation userRep, List<GroupRepresentation> grou
             } else if (groupRep.getName().startsWith("monitor")) {
                 globalOperations.add(Operation.view);
             } else if (groupRep.getName().contains("_")) {
-                String[] parts = groupRep.getName().split("_");
+                String[] parts = groupRep.getName().split("_", 2);
                 Operation operation = Operation.valueOf(parts[0]);
-                String address = decodePart(parts[1]);
+                String address = parts[1];
```

Comment:
I think you still need to decode the part though (see failing test):

```suggestion
                String address = decodePart(parts[1]);
```